spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerOperations.java

```diff
@@ -73,6 +103,18 @@
 	 */
 	String getSecretString(String secretId, String versionName);
 
+	/**
+	 * Gets the secret payload of the specified {@code secretId} at version
+	 * {@code versionName} for a specific {@code projectId}.
+	 *
+	 * @param secretId unique identifier of your secret in Secret Manager.
+	 * @param versionName which version of the secret to load. The version can be a version
+	 *     number as a string (e.g. "5") or an alias (e.g. "latest").
+	 * @param projectId unique identifier of your project.
+	 * @return The secret payload as String
+	 */
+	String getSecretString(String secretId, String versionName, String projectId);
```

Comment:
Are we missing `String getSecretString(String secretId, String projectId)` variation or would this be too much?

Reply:
No, it's not missing. We discussed previously about this.

> Java method signature are on the type, not on the param name and there were a conflict between these (for example):
>
>```
>	@Override
>	public String getSecretString(String secretId, String versionName) {....}
>
>	@Override
>	public String getSecretString(String secretId, String projectId) {....}
>```

So we choose to only propose the projectId with the version. And that's why I also introduced a static public variable `LATEST_VERSION` for helping the developers when they want to get the secret without care about the version

But, we can explore another solution.